### ISSUE TITLE
map is:rft.aucorp to (e.g.) finc:author_corporate

### ISSUE THREAD

**Issue by zazi** [miku/span#12]
currently there is no mapping (in the [intermediate schema to finc Solr schema mapping script](https://github.com/miku/span/blob/master/formats/finc/solr.go)) from intermediate schema field "rft.aucorp" to a field of the finc Solr schema (e.g. finc:author_corporate). can we add a mapping for this field, please?

**Comment by miku** [comment]
Ok, a first take on this is implemented with [0.1.259](https://github.com/miku/span/releases/tag/v0.1.259). Caveat is noted in the [commit](https://github.com/miku/span/commit/0025491776a9a8e0dca9b5ed947ead48960edad4):

> This might be incorrect, as SOLR/VuFind sometimes depends on parallel
filling of separate, but related fields, e.g.:

    author:      ["Alexandros K.", "Almgeirr. Y."]
    author_corp: [""             , "Háskóli Íslands"]

This first implementation just fills author_corps with whatever is available and does not account for any order (if it is significant - I am unsure at the moment).

**Comment by zazi** [comment]
> This first implementation just fills author_corp with whatever is available and does not account for any order (if it is significant - I am unsure at the moment).

no, from my knowledge so far (i.e. as far as I can remember), the position is not relevant, since in our examples in such cases, where we want to map to rft.aucorp, no other author exists in these records. so corporates are not related to authors.

**Comment by zazi** [comment]
I'm just wondering a bit about the naming, I think it's "author_corporate" instead of "author_corp", see, e.g., in the [finc Solr schema definition](https://github.com/finc/index/blob/master/schema.xml#L234). Furthermore, why don't you need to define this field in the [struct of the finc Solr schema in solr.go](https://github.com/miku/span/blob/0025491776a9a8e0dca9b5ed947ead48960edad4/formats/finc/solr.go#L15) as well?

**Comment by miku** [comment]
Thanks for clarifying. It should be a bit better [now](https://github.com/miku/span/releases/tag/v0.1.260).

```shell
$ echo '{"authors": [{"rft.au": "X", "rft.aucorp": "Wayland-Yutani"}]}' | span-export
{
  "access_facet": "Electronic Resources",
  "author_facet": [
    "Wayland"
  ],
  "author_corp": [
    "Wayland-Yutani"
  ],
  "author": [
    "Wayland"
  ],
  "author_sort": "wayland",
  "allfields": "Wayland",
  "facet_avail": [
    "Online"
    ...
```

**Comment by zazi** [comment]
thanks a lot again for your very fast fix here. we'll test it asap.

**Comment by zazi** [comment]
"author_corp" should be "author_corporate", see [this comment](https://github.com/miku/span/commit/988acf91222bde95847de9bccc2a30dcfd6e1008#r30861358)

**Comment by miku** [comment]
Maybe better via https://github.com/miku/span/commit/d358db5df972fbd9e09246681684ed2b242c6060.

**Comment by zazi** [comment]
thanks, we'll it asap.

**Comment by zazi** [comment]
thanks, we'll it asap.

**Comment by zazi** [comment]
it looks like that if no author name or last name or id is given, that author corporate will not be set, see https://github.com/miku/span/blob/b52405ccae5df464113720b34c565dfb3930ddab/formats/finc/intermediate.go#L66 -> i.e. we think that one need to add rft.aucorp as return value to the "toString" method for authors as well.

**Comment by miku** [comment]
> it looks like that if no author name or last name or id is given, that author corporate will not be set, see 

Yes, that's true, `author_corporate` depends on the [authors](https://github.com/miku/span/blob/2aab95496edc0916a5c9ed6520e1d5e23a7e940e/formats/finc/solr.go#L170-L182).

**Comment by zazi** [comment]
any chance to fix the current behavior? we have cases, where only rft.aucorp is filled, i.e., only corporates are part of those records and no "real" authors.

**Comment by bmuschall** [comment]
> Yes, that's true, `author_corporate` depends on the [authors](https://github.com/miku/span/blob/2aab95496edc0916a5c9ed6520e1d5e23a7e940e/formats/finc/solr.go#L170-L182).

author_corporate is independent and should not depend on authors

**Comment by zazi** [comment]
@miku: 

> any chance to fix the current behavior? we have cases, where only rft.aucorp is filled, i.e., only corporates are part of those records and no "real" authors.

any news on this issue? ;)

**Comment by miku** [comment]
Maybe solved in [v0.1.270](https://github.com/miku/span/releases/tag/v0.1.270).

**Comment by zazi** [comment]
looks like that everything works now as expected. thanks a lot.
